lib/spack/spack/package_base.py

```diff
@@ -1947,6 +1916,7 @@ def run_test(
                 in the install prefix bin directory or the provided work_dir
             work_dir (str or None): path to the smoke test directory
         """
+        tty.warn("This method is deprecated.  Use 'test_part' instead.")
```

Comment:
Is:
```suggestion
        warnings.warn("This method is deprecated.  Use 'test_part' instead.")
```
behaving the same? If so it's preferred, since we can change in a unique place what to do with the warning (right now it gets forwarded by default to `tty.warn`)

Reply:
Appears to be, though not for some of the unit testing.  Adjusted tests accordingly.